### ISSUE TITLE
Missing varargs cast to remove warning

### ISSUE THREAD

**Issue by FroMage** [ceylon/ceylon-compiler#1822]
``` ceylon
shared class AnythingVoidHandler(void done(Anything n))  {
  shared void handle() {
    done("");
  }
}
```

Produces:

```
test/src/com/redhat/ceylon/compiler/java/test/issues/bug18xx/Bug18XX.ceylon:20:WARNING:non-varargs call of varargs method with inexact argument type for last parameter;
  cast to java.lang.Object for a varargs call
  cast to java.lang.Object[] for a non-varargs call and to suppress this warning
```
